esphome/__main__.py

```diff
@@ -140,6 +140,12 @@ def choose_upload_log_host(
                     resolved.append("MQTT")
             else:
                 resolved.append(device)
+
+        resolved = [i for i in resolved if i is not None and len(i) > 0]
```

Comment:
None and empty strings are both falsey

```suggestion
        resolved = [i for i in resolved if i]
```